packages/block-library/src/navigation/edit/index.js

```diff
@@ -642,11 +620,13 @@ function Navigation( {
 								hasResolvedNavigationMenus
 							}
 							clientId={ clientId }
+							canUserCreateNavigation={ canUserCreateNavigation }
 						/>
 					) }
-					{ ! isEntityAvailable && ! isPlaceholderShown && (
-						<PlaceholderPreview isLoading />
-					) }
+					{ ! hasResolvedCanUserCreateNavigation ||
+						( ! isEntityAvailable && ! isPlaceholderShown && (
```

Comment:
This condition is becoming long and hard to unwind. I found it confusing that we're showing a placeholder when `! isPlaceholderShown`. It's not specific to this PR, but it would be nice to have a `<LoadingState />` component or so.

Reply:
What should I change here? Or is this for a followup?

Reply:
Nothing to change in this PR, I'm just making a note for the future. I added a mention into the code quality issue https://github.com/WordPress/gutenberg/issues/37190